core/src/main/scala/org/apache/spark/sql/CachedDataFrame.scala

```diff
@@ -133,6 +131,11 @@ class CachedDataFrame(df: Dataset[Row], var queryString: String,
    */
   private def withCallback[U](name: String)(action: DataFrame => U) = {
     try {
+      val snSession = SparkSession.getActiveSession.get.asInstanceOf[SnappySession]
+      val pool = snSession.asInstanceOf[SnappySession].
+        sessionState.conf.activeSchedulerPool
+
+      snSession.sparkContext.setLocalProperty("spark.scheduler.pool", pool)
```

Comment:
Can we set lowlatency by default for pruned tasks? That will be very useful so users will not have to keep setting it explicitly. I think it can be something like:

```scala
if ((cachedRDD ne null) && cachedRDD.getNumPartitions <= 2 /* some small number */ &&
    shuffleDependencies.length == 0) && pool == "default") {
  pool = "lowlatency" // perhaps also check that this pool exists and user has not renamed/removed it
}
```

We don't expect lowlatency pool to be used for queries that can launch subqueries, or run jobs in plan execute itself, so overriding default for just the collect should be enough.